docs/framework/guides/creating-simple-plugin-timestamp.md

```diff
@@ -0,0 +1,271 @@
+
+```js
+import ButtonView from '@ckeditor/ckeditor5-ui/src/button/buttonview';
+
+class Timestamp extends Plugin {
+	init() {
+		const editor = this.editor;
+
+		// The button must be registered among the UI components of the editor
+		// to be displayed in the toolbar.
+		editor.ui.componentFactory.add( 'timestamp', locale => {
+			
+			// The button will be an instance of ButtonView
+			const button = new ButtonView( locale );
+
+			button.set( {
+				label: 'Timestamp',
+				withText: true
+			} );
+
+			return button;
+		} );
+	}
+}
+
+ClassicEditor
+	.create( document.querySelector( '#editor' ), {
+		plugins: [ 
+			Timestamp, Essentials, Paragraph, Heading, List, Bold, Italic 
+			], 
+
+		//Add the Timestamp button to the config.toolbar array
+		toolbar: [ 
+			'timestamp', 'heading', 'bold', 'italic', 'numberedList', 'bulletedList'
+			] 
+	} )
+	.then( editor => {
+		console.log( 'Editor was initialized', editor );
+	} )
+	.catch( error => {
+		console.error( error.stack );
+	} );
+
+```
+You should be able to see the timestamp button now. It doesn't do anything just yet, so let's change that. 
+
+## Inserting a timestamp
+
+We can now define the core funcitonality of our plugin, the action that should be executed once our button is clicked. 
+
+When we want to insert something into the document structure (or change it in any other way), we need to use the model writer, available in the model's `change()` method. 
+
+<info-box>
+	What is the model? It's a DOM-like structure, that is converted into the view, which is what the user interacts with. If you want to learn more, you can read about it in the {@link framework/guides/architecture/editing-engine#overview editing engine architecture}. Don't be intimidated by the diagram you can find there, as you don't need to understand it in its full complexity to implement custom plugins.   
+</info-box>
+
+We'll use the {link module:engine/model/writer~Writer#insertText `writer.insertText()`} method to insert our timestamp into the document. We'll also need to give it a position of the user's current selection to indicate where to insert our timestamp (using the {@link module:engine/model/position~Position `Position`} class). 
+
+Finally, if the user's selection has a range (so it's a letter, word, or a whole text fragment), we'll remove that and replace it with our timestamp.
+
+```js
+class Timestamp extends Plugin {
+	init() {
+		const editor = this.editor;
+
+		editor.ui.componentFactory.add( 'timestamp', locale => {
+			
+			//...
+
+			//Execute a callback function when the button is clicked
+			button.on( 'execute', () => {
+				const now = new Date();
+				const selection = editor.model.document.selection;
+
+				//Change the model using the model writer
+				editor.model.change( writer => {
+
+					//Insert the text at the user's current position
+					writer.insertText( `${ now.toString() }`, selection.getFirstPosition() );
```

Comment:
```suggestion
					writer.insertText( now.toString(), selection.getFirstPosition() );
```

Reply:
Anyway, let's use `insertContent` instead because it is simpler and hi-level 

```
editor.model.change( writer => {
	editor.model.insertContent( writer.createText( 'x' ) );
} );
```